R/model.R

```diff
@@ -369,22 +374,146 @@ compile_method <- function(quiet = TRUE,
   if (is.null(include_paths) && !is.null(private$precompile_include_paths_)) {
     include_paths <- private$precompile_include_paths_
   }
-  if (is.null(dir) && !is.null(private$dir_)) {
-    dir <- absolute_path(private$dir_)
-  } else if (!is.null(dir)) {
-    dir <- absolute_path(dir)
-  }
-  if (!is.null(dir)) {
-    dir <- repair_path(dir)
-    checkmate::assert_directory_exists(dir, access = "rw")
-  }
 
   # temporary deprecation warnings
   if (isTRUE(threads)) {
     warning("'threads' is deprecated. Please use 'cpp_options = list(stan_threads = TRUE)' instead.")
     cpp_options[["stan_threads"]] <- TRUE
   }
 
+  model_name <- sub(" ", "_", paste0(strip_ext(basename(self$stan_file())), "_model"))
+
+  stancflags_val <- NULL
+  if (!is.null(include_paths)) {
+    checkmate::assert_directory_exists(include_paths, access = "r")
+    include_paths <- absolute_path(include_paths)
+    include_paths <- paste0(include_paths, collapse = ",")
+    if (cmdstan_version() >= "2.24") {
+      include_paths_flag <- " --include-paths="
+    } else {
+      include_paths_flag <- " --include_paths="
+    }
+    stancflags_val <- paste0(stancflags_val, include_paths_flag, include_paths, " ")
+  }
+
+  if (!is.null(cpp_options$stan_opencl)) {
+    stanc_options[["use-opencl"]] <- TRUE
+  }
+  if (is.null(stanc_options[["name"]])) {
+    stanc_options[["name"]] <- model_name
+  }
+  if(make_or_stanc=='make'){
+    char_for_stanc_built_options <- "'"
+  }else{
+    char_for_stanc_built_options <- ""
+  }
+
+  stanc_built_options = c()
+  for (i in seq_len(length(stanc_options))) {
+    option_name <- names(stanc_options)[i]
+    if (isTRUE(as.logical(stanc_options[[i]]))) {
+      stanc_built_options = c(stanc_built_options, paste0("--", option_name))
+    } else {
+      stanc_built_options = c(stanc_built_options, paste0("--", option_name, "=",
+        char_for_stanc_built_options, stanc_options[[i]], char_for_stanc_built_options))
+    }
+  }
+
+  if (cmdstan_version() < "2.24") {
+    prepare_precompiled(cpp_options, quiet)
+  }
+
+  if(make_or_stanc=='make'){
+    stancflags_val <- paste0("STANCFLAGS += ", stancflags_val, paste0(stanc_built_options, collapse = " "))
+    these_args = c(
+      cpp_options_to_compile_flags(cpp_options),
+      stancflags_val
+    )
+  }else{
+    stancflags_val <- trimws(c(stancflags_val, c(stanc_built_options)))
+    these_args = c(stancflags_val)
+  }
+  return(these_args)
+}
+CmdStanModel$set("public", name = "generate_arg_strings", value = generate_arg_strings_method)
```

Comment:
Does this method need to be public? Is the user ever going to call it themselves?

Reply:
Oh, nope, could totally be private. I just was copying existing code and didn't catch that bit.

Reply:
Ok cool. I think we should make it private if it won't be called by the user. Oh and if you switch that to private you'll need to use `private$generate_arg_strings` instead of `self$generate_arg_strings` when calling it. I had that issue once in the past and it took me an embarrassingly long time to figure out why `self$foo` wasn't working anymore.